### ISSUE TITLE
tendrl-node-agent fails to start after reboot

### ISSUE THREAD

**Issue by ltrilety** [Tendrl/node-agent#373]
Sometimes when machine is rebooted `tendrl-node-agent` fails to start.

```
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: Finding flows in namespace.node_agent.flowsSetup Tendrl definitions (.yml) for namespace.node_agentValidating registered (.py) objects in namespace.node_agent.objectsValidating defined (.yml) objects in namespace.node_agent.objectsSetup Config for namespace.node_agentSetup Etcd Orm for namespace.node_agentTraceback (most recent call last):
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: File "/usr/bin/tendrl-node-agent", line 9, in <module>
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: load_entry_point('tendrl-node-agent==1.2.2', 'console_scripts', 'tendrl-node-agent')()
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: File "/usr/lib/python2.7/site-packages/tendrl/node_agent/manager/__init__.py", line 40, in main
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: node_agent.NodeAgentNS()
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: File "/usr/lib/python2.7/site-packages/tendrl/node_agent/__init__.py", line 14, in __init__
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: super(NodeAgentNS, self).__init__(ns_name, ns_src)
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: File "/usr/lib/python2.7/site-packages/tendrl/commons/__init__.py", line 63, in __init__
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: self.setup_common_objects()
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: File "/usr/lib/python2.7/site-packages/tendrl/commons/__init__.py", line 430, in setup_common_objects
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: NS.etcd_orm = etcdobj.Server(etcd_kwargs=etcd_kwargs)
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: File "/usr/lib/python2.7/site-packages/tendrl/commons/etcdobj/__init__.py", line 161, in __init__
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: etcd.Client(**etcd_kwargs))
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: File "/usr/lib/python2.7/site-packages/etcd/client.py", line 202, in __init__
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: set(self.machines))
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: File "/usr/lib/python2.7/site-packages/etcd/client.py", line 304, in machines
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: raise etcd.EtcdException("Could not get the list of servers, "
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com tendrl-node-agent[1316]: etcd.EtcdException: Could not get the list of servers, maybe you provided the wrong host(s) to connect to?
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com systemd[1]: tendrl-node-agent.service: main process exited, code=exited, status=1/FAILURE
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com systemd[1]: Unit tendrl-node-agent.service entered failed state.
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com systemd[1]: tendrl-node-agent.service failed.
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com systemd[1]: tendrl-node-agent.service holdoff time over, scheduling restart.
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com systemd[1]: start request repeated too quickly for tendrl-node-agent.service
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com systemd[1]: Failed to start A python agent local to every managed storage node in the sds cluster.
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com systemd[1]: Unit tendrl-node-agent.service entered failed state.
Apr 10 12:47:24 ltrilety-usm1-gl4.usmqe.lab.eng.brq.redhat.com systemd[1]: tendrl-node-agent.service failed.
```

Used packages:
tendrl-commons-1.2.2-04_10_2017_04_42_05.noarch
tendrl-node-agent-1.2.2-04_10_2017_00_01_03.noarch
tendrl-node-monitoring-1.2.2-04_10_2017_16_13_14.noarch

**Comment by ltrilety** [comment]
My bad, the order of starting machines was not fixed hence the etcd connection was really not available. However `node-agent` fails to start on the server itself, when the server is rebooted, with the same error.

**Comment by shtripat** [comment]
@ltrilety the error " etcd.EtcdException: Could not get the list of servers, maybe you provided the wrong host(s) to connect to?" comes if the URl of etcd server is not correct in `/etc/tendrl/node-agent.conf.yaml`. Not sure how that gets changed afetr re-start of the nodes. Ideally it would happen if `tendrl-node-agent` re-install/update happens. This is tracked under https://github.com/Tendrl/node-agent/issues/364 and seems to be fixed.

**Comment by ltrilety** [comment]
@shtripat sorry, the issue is valid only for server. It seems there's bad order of starting services, looks like the etcd was started after attempt to start `node-agent` and `performance-monitoring` is already made.

**Comment by ltrilety** [comment]
@shtripat  I was thinking about the issue and I don't feel it's a correct behavior to fail to start a service when some connection is missing. The service should start and tried regularly if the connection is available.
That said I tried a simple scenario. I shutdown tendrl server and tried to stop/start any tendrl service on any node. Surprisingly it seems to work as it should. Service was started. There was similar `etcd` error in the log, but service was active and not failed to start. When the connection is available it uses it correctly.
So the issue is related somehow to `etcd` connection, but that's not only think which is bad here.

**Comment by ltrilety** [comment]
good news is that after some time systemd itself without user intervention was able to start the service

**Comment by r0h4n** [comment]
The dependency on etcd systemd service should be in all .service descriptors of tendrl services, ill make that change. 

This will ensure etcd is indeed started before any tendrl service

**Comment by r0h4n** [comment]
Fixed: https://github.com/Tendrl/node-agent/commit/57ea0f7701765d459bcc964b8ce9035f48fa450d

Let me know if the issue still persists

**Comment by ltrilety** [comment]
The issue seems to be still valid.
I restarted my machines and on some of them `tendrl-node-agent` fails to start (BTW the same for `tendrl-node/performance-monitoring`).

Logs, and configurations:
[evidence_files.zip](https://github.com/Tendrl/node-agent/files/928092/evidence_files.zip)

Checked on:
tendrl-node-monitoring-1.2.2-04_18_2017_04_25_03.noarch
tendrl-commons-1.2.2-04_17_2017_20_19_16.noarch
tendrl-node-agent-1.2.3-04_18_2017_02_22_22.noarch
tendrl-api-httpd-1.2.2-04_18_2017_01_51_04.noarch
tendrl-api-doc-1.2.2-04_18_2017_01_51_04.noarch
tendrl-commons-1.2.2-04_17_2017_20_19_16.noarch
tendrl-dashboard-1.2.1-04_18_2017_04_11_04.noarch
tendrl-performance-monitoring-1.2.2-04_18_2017_03_02_04.noarch
etcd-3.1.0-2.el7.x86_64
tendrl-api-1.2.2-04_18_2017_01_51_04.noarch

**Comment by r0h4n** [comment]
Just so there is no confusion, the expectation here is that the tendrl-node-agent should start successfully on node reboot

**Comment by GowthamShanmugam** [comment]
@ltrilety @r0h4n @shtripat There is two kind of issue,
1.problem in other nodes (except etcd node) when etcd node is  restarted 
   if we use firewalld disable command 
      service firewalld stop , then it will temporarily stops the firewalld, if we reboot etcd node then again firewalld service will enabled. Because firewalld other nodes can't communicate with etcd. so it gives error like   
etcd.EtcdException: Could not get the list of servers, maybe you provided the wrong host(s) to connect to?

so systemctl disable firewalld will stop firewalld permanently.
2. problem in etcd node :
    if we restart etcd node then node-agent service  in etcd node is failing, but etcd is in running state only.
   it happening because node-agent service start before etcd start. i gave keyword After=etcd.service in node-agent service then it works fine. but we can't give etcd as dependency. i tested like this for confirm the issue. 
@TimothyAsir has patch for this problem

**Comment by TimothyAsirJeyasing** [comment]
etcd starts after network is available. So any service started after etcd will have network services.
The actual cause of this issue is just because of network service not available.

**Comment by shtripat** [comment]
@GowthamShanmugam regarding point-1 in https://github.com/Tendrl/node-agent/issues/373#issuecomment-295201992 stopping firewalld is not the solution.
Ideally the required ports on nodes should be opened so that they can communicate.

Please figure out what port nos are required for etcd communication from storage nodes and on etcd node and lets ask to open those ports before starting the services.

**Comment by GowthamShanmugam** [comment]
@shtripat ok

**Comment by GowthamShanmugam** [comment]
@TimothyAsir @shtripat I have one doubt here if network is the problem, why other nodes node-agent service is not fail.

**Comment by ltrilety** [comment]
@TimothyAsir @shtripat @GowthamShanmugam I have to put here my note too. When the `etcd` server is switched off it doesn't mean that `node-agent` or `monitoring` service cannot be started. E.g. I tried the scenario where I turned off the tendrl machine and tried to stop and start `node-agent` on some node machine and it was able to start. The same error message about `etcd` was present but `node-agent` was running. And that is what should happen.

**Comment by GowthamShanmugam** [comment]
@ltrilety please check the status of firewalld when you face this error

**Comment by ltrilety** [comment]
@GowthamShanmugam I have disabled `firewalld` all the time

**Comment by r0h4n** [comment]
@ltrilety Can you reproduce this one? I am not sure the copr tendrl/tendrl repo builds have this issue

**Comment by ltrilety** [comment]
Seems to be working, I was not able to reproduce it. 

Checked with:
tendrl-api-httpd-1.2.3-05_04_2017_01_51_04.noarch
tendrl-dashboard-1.2.3-05_04_2017_04_11_06.noarch
tendrl-performance-monitoring-1.2.3-05_04_2017_11_57_30.noarch
tendrl-api-doc-1.2.3-05_04_2017_01_51_04.noarch
tendrl-node-agent-1.2.3-05_04_2017_11_40_12.noarch
etcd-3.1.3-1.el7.x86_64
tendrl-api-1.2.3-05_04_2017_01_51_04.noarch
tendrl-commons-1.2.3-05_04_2017_05_13_04.noarch

**Comment by ltrilety** [comment]
After several other checks I was able to reproduce that.
I switch off all machines. Then I was starting them again. I put there some sleep between starting of machines and tendrl machine itself. All machines except tendrl one were started and after 60 seconds (could be probably less) tendrl machine started. With that on some machines node-agent failed to start not on all of them though.

**Comment by r0h4n** [comment]
> With that on some machines node-agent failed to start not on all of them though

Please provide logs where node-agent failed to start

**Comment by ltrilety** [comment]
logs and configuration:
[evidence-files.zip](https://github.com/Tendrl/node-agent/files/978563/evidence-files.zip)

E.g. `node-agent` is not running on `ltrilety-usm1-gl2.usmqe.lab.eng.brq.redhat.com`

**Comment by r0h4n** [comment]
Fixed, let us know if the issue persists

**Comment by mkudlej** [comment]
I think this issue it still there, this is status of tendrl services after nodes restart:
```
ansible -i ./mkudlej-usm2.hosts -m shell -a 'systemctl -a | grep tendrl' all
mkudlej-usm2-osd2. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-osd1. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-mon3. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-mon1. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-mon2. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-osd3. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-gl2. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-gl1. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-osd4. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-gl3. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-gl4. | SUCCESS | rc=0 >>
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-node-monitoring.service                                              loaded    failed   failed    Daemon to manage tendrl node monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging

mkudlej-usm2-server. | SUCCESS | rc=0 >>
  tendrl-apid.service                                                         loaded    active   running   Tendrl Api Daemon
● tendrl-node-agent.service                                                   loaded    failed   failed    A python agent local to every managed storage node in the sds cluster
● tendrl-performance-monitoring.service                                       loaded    failed   failed    Daemon to manage tendrl monitoring
  tendrl-node-agent.socket                                                    loaded    active   listening Tendrl message socket for logging
```
```systemctl status tendrl-node-agent``` for one node and server:
```
mkudlej-usm2-gl4. | FAILED | rc=3 >>
● tendrl-node-agent.service - A python agent local to every managed storage node in the sds cluster
   Loaded: loaded (/usr/lib/systemd/system/tendrl-node-agent.service; enabled; vendor preset: disabled)
   Active: failed (Result: start-limit) since Tue 2017-05-09 09:03:37 UTC; 16min ago
  Process: 1988 ExecStart=/usr/bin/tendrl-node-agent (code=exited, status=1/FAILURE)
 Main PID: 1988 (code=exited, status=1/FAILURE)

May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 626, in resolve
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: raise DistributionNotFound(req)
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: pkg_resources.DistributionNotFound: dnspython>=1.13.0
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service failed.
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service holdoff time over, scheduling restart.
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: start request repeated too quickly for tendrl-node-agent.service
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: Failed to start A python agent local to every managed storage node in the sds cluster.
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service failed.

mkudlej-usm2-server. | FAILED | rc=3 >>
● tendrl-node-agent.service - A python agent local to every managed storage node in the sds cluster
   Loaded: loaded (/usr/lib/systemd/system/tendrl-node-agent.service; enabled; vendor preset: disabled)
   Active: failed (Result: start-limit) since Tue 2017-05-09 09:03:35 UTC; 16min ago
  Process: 1580 ExecStart=/usr/bin/tendrl-node-agent (code=exited, status=1/FAILURE)
 Main PID: 1580 (code=exited, status=1/FAILURE)

May 09 09:03:35 mkudlej-usm2-server. systemd[1]: tendrl-node-agent.service: main process exited, code=exited, status=1/FAILURE
May 09 09:03:35 mkudlej-usm2-server. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:35 mkudlej-usm2-server. systemd[1]: tendrl-node-agent.service failed.
May 09 09:03:35 mkudlej-usm2-server. systemd[1]: tendrl-node-agent.service holdoff time over, scheduling restart.
May 09 09:03:35 mkudlej-usm2-server. systemd[1]: start request repeated too quickly for tendrl-node-agent.service
May 09 09:03:35 mkudlej-usm2-server. systemd[1]: Failed to start A python agent local to every managed storage node in the sds cluster.
May 09 09:03:35 mkudlej-usm2-server. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:35 mkudlej-usm2-server. systemd[1]: tendrl-node-agent.service failed.
```
```journalctl -u tendrl-node-agent``` on one node:
```
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: Started A python agent local to every managed storage node in the sds cluster.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: Starting A python agent local to every managed storage node in the sds cluster...
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[806]: Traceback (most recent call last):
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[806]: File "/usr/bin/tendrl-node-agent", line 5, in <module>
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[806]: from pkg_resources import load_entry_point
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[806]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 3011, in <module>
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[806]: parse_requirements(__requires__), Environment()
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[806]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 626, in resolve
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[806]: raise DistributionNotFound(req)
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[806]: pkg_resources.DistributionNotFound: dnspython>=1.13.0
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service: main process exited, code=exited, status=1/FAILURE
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service failed.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service holdoff time over, scheduling restart.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: Started A python agent local to every managed storage node in the sds cluster.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: Starting A python agent local to every managed storage node in the sds cluster...
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1011]: Traceback (most recent call last):
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1011]: File "/usr/bin/tendrl-node-agent", line 5, in <module>
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1011]: from pkg_resources import load_entry_point
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1011]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 3011, in <module>
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1011]: parse_requirements(__requires__), Environment()
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1011]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 626, in resolve
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1011]: raise DistributionNotFound(req)
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1011]: pkg_resources.DistributionNotFound: dnspython>=1.13.0
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service: main process exited, code=exited, status=1/FAILURE
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service failed.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service holdoff time over, scheduling restart.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: Started A python agent local to every managed storage node in the sds cluster.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: Starting A python agent local to every managed storage node in the sds cluster...
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1384]: Traceback (most recent call last):
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1384]: File "/usr/bin/tendrl-node-agent", line 5, in <module>
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1384]: from pkg_resources import load_entry_point
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1384]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 3011, in <module>
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1384]: parse_requirements(__requires__), Environment()
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1384]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 626, in resolve
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1384]: raise DistributionNotFound(req)
May 09 09:03:35 mkudlej-usm2-gl4. tendrl-node-agent[1384]: pkg_resources.DistributionNotFound: dnspython>=1.13.0
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service: main process exited, code=exited, status=1/FAILURE
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:35 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service failed.
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service holdoff time over, scheduling restart.
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: Started A python agent local to every managed storage node in the sds cluster.
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: Starting A python agent local to every managed storage node in the sds cluster...
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1804]: Traceback (most recent call last):
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1804]: File "/usr/bin/tendrl-node-agent", line 5, in <module>
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1804]: from pkg_resources import load_entry_point
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1804]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 3011, in <module>
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1804]: parse_requirements(__requires__), Environment()
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1804]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 626, in resolve
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1804]: raise DistributionNotFound(req)
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1804]: pkg_resources.DistributionNotFound: dnspython>=1.13.0
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service: main process exited, code=exited, status=1/FAILURE
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service failed.
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service holdoff time over, scheduling restart.
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: Started A python agent local to every managed storage node in the sds cluster.
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: Starting A python agent local to every managed storage node in the sds cluster...
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: Traceback (most recent call last):
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: File "/usr/bin/tendrl-node-agent", line 5, in <module>
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: from pkg_resources import load_entry_point
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 3011, in <module>
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service: main process exited, code=exited, status=1/FAILURE
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: parse_requirements(__requires__), Environment()
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 626, in resolve
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: raise DistributionNotFound(req)
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: pkg_resources.DistributionNotFound: dnspython>=1.13.0
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:36 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service failed.
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service holdoff time over, scheduling restart.
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: start request repeated too quickly for tendrl-node-agent.service
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: Failed to start A python agent local to every managed storage node in the sds cluster.
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: Unit tendrl-node-agent.service entered failed state.
May 09 09:03:37 mkudlej-usm2-gl4. systemd[1]: tendrl-node-agent.service failed.
```

**Comment by r0h4n** [comment]
```
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: parse_requirements(__requires__), Environment()
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 626, in resolve
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: raise DistributionNotFound(req)
May 09 09:03:36 mkudlej-usm2-gl4. tendrl-node-agent[1988]: pkg_resources.DistributionNotFound: dnspython>=1.13.0
```

duplicate of https://github.com/Tendrl/node-agent/issues/409

**Comment by ltrilety** [comment]
@r0h4n Something is not right here. Correct me if I am wrong.
The https://github.com/Tendrl/node-agent/issues/409 issue was closed as fixed whilst in the same time You said that the issue commented by @mkudlej here is duplicate of this issue. Are You aware that @mkudlej has the newest available packages. That said there are only two possible explanations first it's not the same issue or second the issue is not fixed.

**Comment by r0h4n** [comment]
The logs provided by @mkudlej (here : https://github.com/Tendrl/node-agent/issues/373#issuecomment-300110024) are about the python-dns version issue. Hence I had called the comment a duplicate of https://github.com/Tendrl/node-agent/issues/409


anyways, does the tendrl-node-agent still fails to start after reboot ?

**Comment by ltrilety** [comment]
@r0h4n sorry, I didn't check that before as @mkudlej contacted me with his settings where he has problem which looks like this one. However I checked that now and it seems to be working I didn't detected any node-agent which failed to start after reboot.
However from https://github.com/Tendrl/node-agent/issues/373#issuecomment-300110024 https://github.com/Tendrl/node-agent/issues/409 seems still valid. Maybe there's different scenario to hit that though.

**Comment by mkudlej** [comment]
Tested with 
```
centos-release-ceph-jewel-1.0-1.el7.centos.noarch
ceph-ansible-2.2.0-90.g31267ab.el7.noarch
ceph-installer-1.3.0-5.g8ec17ca.el7.noarch
etcd-3.1.3-1.el7.x86_64
python-etcd-0.4.5-1.noarch
rubygem-etcd-0.3.0-1.el7.centos.noarch
tendrl-api-1.2.3-05_10_2017_01_51_05.noarch
tendrl-api-doc-1.2.3-05_10_2017_01_51_05.noarch
tendrl-api-httpd-1.2.3-05_10_2017_01_51_05.noarch
tendrl-commons-1.2.3-05_10_2017_05_13_05.noarch
tendrl-dashboard-1.2.3-05_10_2017_04_11_05.noarch
tendrl-node-agent-1.2.3-05_10_2017_00_01_05.noarch
tendrl-node-monitoring-1.2.3-05_10_2017_04_25_07.noarch
tendrl-performance-monitoring-1.2.3-05_10_2017_03_02_05.noarch
```
and it works.
